cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -799,7 +801,7 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onPendingProposal(
                             p.fail(e);
                         });
             } else {
-                apiClient.getUserTaskStatus(reconciliation, host, cruiseControlPort, sessionId)
+                fromCompletableFuture(apiClient.getUserTaskStatus(reconciliation, host, cruiseControlPort, sessionId))
```

Comment:
Can't you use `Future.fromCompletionStage` instead? I'm not sure if you considered it and decided against it for some reason or not when writing the custom method for it.

Reply:
I did use fromCompletionStage initially but on failure, the Future is completed with CompletionException, rather than the root cause. So I created this method to unwrap the CompletionException and fail the future with the root cause.

Reply:
Ok, that makes sense. Thanks for the explanation. It might be worth to add this to the JavaDoc of the custom method you created for handling this.